cmd/cost/get.go

```diff
@@ -0,0 +1,244 @@
+package cost
+
+import (
+	"fmt"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/costexplorer"
+	"github.com/aws/aws-sdk-go/service/organizations"
+
+	"github.com/spf13/cobra"
+
+	awsprovider "github.com/openshift/osd-utils-cli/pkg/provider/aws"
+)
+
+// getCmd represents the get command
+func newCmdGet(streams genericclioptions.IOStreams) *cobra.Command {
+	ops := newGetOptions(streams)
+	getCmd := &cobra.Command{
+		Use:   "get",
+		Short: "Get total cost of a given OU. If no OU given, then gets total cost of v4 OU.",
+		Run: func(cmd *cobra.Command, args []string) {
+
+			cmdutil.CheckErr(opsCost.complete(cmd, args))
+			org, ce, err := opsCost.initAWSClients()
+			cmdutil.CheckErr(err)
+
+			//Get Organizational Unit
+			OU := organizations.OrganizationalUnit{Id: aws.String(ops.ou)}
+			//Store cost
+			var cost float64 = 0
+
+			if ops.recursive {
+				getOUCostRecursive(&OU, org, ce, &ops.time, &cost)
+				fmt.Printf("Cost of %s recursively is: %f\n", ops.ou, cost)
+			} else {
+				getOUCost(&OU, org, ce, &ops.time, &cost)
+				fmt.Printf("Cost of %s is: %f\n", ops.ou, cost)
+			}
+		},
+	}
+	getCmd.Flags().StringVar(&ops.ou, "ou", "ou-0wd6-aff5ji37", "get OU ID (default is v4)") //Default OU is v4
+	getCmd.Flags().BoolVarP(&ops.recursive, "recursive", "r", false, "recurse through OUs")
+	getCmd.Flags().StringVarP(&ops.time, "time", "t", "", "set time")
+
+	return getCmd
+}
+
+//Store flag options for get command
+type getOptions struct {
+	ou        string
+	recursive bool
+	time      string
+
+	genericclioptions.IOStreams
+}
+
+func newGetOptions(streams genericclioptions.IOStreams) *getOptions {
+	return &getOptions{
+		IOStreams: streams,
+	}
+}
+
+//Get account IDs of immediate accounts under given OU
+func getAccounts(OU *organizations.OrganizationalUnit, org awsprovider.OrganizationsClient) []*string {
+	//accountSlice stores accounts
+	var accountSlice []*string
+
+	//Get accounts
+	accounts, err := org.ListAccountsForParent(&organizations.ListAccountsForParentInput{
+		ParentId: OU.Id,
+	})
+
+	//Populate accountSlice with accounts by looping until accounts.NextToken is null
+	for {
+		if err != nil { //Look at this for error handling: https://docs.aws.amazon.com/sdk-for-go/api/service/organizations/#example_Organizations_ListOrganizationalUnitsForParent_shared00
+			log.Fatalln("Unable to retrieve accounts under OU:", err)
+		}
+
+		for i := 0; i < len(accounts.Accounts); i++ {
+			accountSlice = append(accountSlice, accounts.Accounts[i].Id)
+		}
+
+		if accounts.NextToken == nil {
+			break
+		}
+
+		//Get accounts
+		accounts, err = org.ListAccountsForParent(&organizations.ListAccountsForParentInput{
+			ParentId:  OU.Id,
+			NextToken: accounts.NextToken,
+		})
+	}
```

Comment:
good start here, but I think we can make this logic flow better. currently you have 2 ListAccountsForParents calls. Its possible to only do this once by putting the call in line 74 inside the for loop and set the first token to 0. Here is an example of pagination in another place https://github.com/openshift/aws-account-operator/blob/master/pkg/totalaccountwatcher/totalaccountwatcher.go#L108 PTAL and reach out if you need more help fixing this